# Copyright (C) 2021 Alexandros Theodotou <alex at zrythm dot org>
#
# This file is part of Zrythm
#
# Zrythm is free software: you can redistribute it and/or modify
# it under the terms of the GNU Affero General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# Zrythm is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Affero General Public License for more details.
#
# You should have received a copy of the GNU Affero General Public License
# along with Zrythm.  If not, see <https://www.gnu.org/licenses/>.

diff --git a/src/cairo-surface.c b/src/cairo-surface.c
index 86df1272e..0a95047f3 100644
--- a/src/cairo-surface.c
+++ b/src/cairo-surface.c
@@ -2174,6 +2174,7 @@ _cairo_surface_paint (cairo_surface_t             *surface,
                      const cairo_clip_t        *clip)
 {
     cairo_int_status_t status;
+    cairo_bool_t is_clear;

     TRACE ((stderr, "%s\n", __FUNCTION__));
     if (unlikely (surface->status))
@@ -2196,8 +2197,9 @@ _cairo_surface_paint (cairo_surface_t             *surface,
        return status;

     status = surface->backend->paint (surface, op, source, clip);
-    if (status != CAIRO_INT_STATUS_NOTHING_TO_DO) {
-       surface->is_clear = op == CAIRO_OPERATOR_CLEAR && clip == NULL;
+    is_clear = op == CAIRO_OPERATOR_CLEAR && clip == NULL;
+    if (status != CAIRO_INT_STATUS_NOTHING_TO_DO || is_clear) {
+       surface->is_clear = is_clear;
        surface->serial++;
     }

